packages/@webex/plugin-cc/src/AgentProfile/AgentProfile.ts

```diff
@@ -0,0 +1,108 @@
+import {AgentProfileResponse} from './types';
+import AgentProfileService from '../AgentProfileService/AgentProfileService';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  ListTeamsResponse,
+  UserResponse,
+} from '../AgentProfileService/types';
+import {WebexSDK} from '../types';
+import {attributes, page, pageSize} from '../AgentProfileService/constants';
+
+export default class AgentProfile {
+  ciUserId: string;
+  agentProfile: AgentProfileResponse;
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(ciUserId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.webex = webex;
+    this.ciUserId = ciUserId;
+    this.agentProfile = {
+      userDetails: {} as UserResponse['body'],
+      agentDesktopProfile: {} as DesktopProfileResponse['body'],
+      teamsList: [] as ListTeamsResponse['body'],
+      auxCodesList: [] as ListAuxCodesResponse['body']['data'],
+    };
+    this.wccAPIURL = wccAPIURL;
+  }
+
+  /**
+   * Method to get Agent Profile by providing ciUserId.
+   * @param {string} ciUserId The CI ID of a User.
+   * @returns {Promise<AgentProfileResponse>} A promise that eventually resolves to an API response and return configuration of an Agent.
+   */
+
+  public async getAgentProfile(ciUserId: string): Promise<AgentProfileResponse> {
+    try {
+      const orgId = await this.webex.credentials.getOrgId();
+      if (!ciUserId || !orgId) {
+        this.webex.logger.error(`Please provide ciUserId and orgId`);
+        Promise.reject(new Error('Please provide ciUserId and orgId'));
+      }
+
+      const agentProfileService = new AgentProfileService(
+        ciUserId,
+        orgId,
+        this.webex,
+        this.wccAPIURL
+      );
+
+      const user: UserResponse = await agentProfileService.getUserUsingCI(ciUserId, orgId);
+      this.agentProfile.userDetails = {
+        agentProfileId: user?.body?.agentProfileId || '',
+        teamIds: user?.body?.teamIds || [''],
+        userProfileId: user?.body?.userProfileId || '',
+      };
+
+      const agentDesktopProfile: DesktopProfileResponse =
+        await agentProfileService.retrieveDesktopProfileById(orgId, user?.body?.agentProfileId);
+      this.agentProfile.agentDesktopProfile = {
+        buddyTeams: agentDesktopProfile?.body?.buddyTeams || [''],
+        idleCodes: agentDesktopProfile?.body?.idleCodes || [''],
+        wrapUpCodes: agentDesktopProfile?.body?.wrapUpCodes || [''],
+        loginVoiceOptions: agentDesktopProfile?.body?.loginVoiceOptions || [''],
+        queues: agentDesktopProfile?.body?.queues || [''],
```

Comment:
Optional fields are usually the last resort.. why do we have so many optional fields?

Reply:
I have used the optional fields because it is bad to show undefined to the developer who are using our CC SDK rather than we should show an empty array or string. It might be the case from CC developer portal API that for some users these fields are not present or might give empty array so to handle that I have given optional fields otherwise it will show undefined.

Reply:
CC developer portal does not have document around this. I have corrected the code.

Reply:
As discussed in the scrum, we are going forward to have these fields mandatory. Code Updated!